core/templates/dev/head/domain/story_viewer/ReadOnlyStoryNodeObjectFactory.ts

```diff
@@ -17,72 +17,88 @@
-  };
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-  // Instance methods
+export class StoryNode {
+  id;
+  title;
+  destinationNodeIds;
+  prerequisiteSkillIds;
+  acquiredSkillIds;
+  outline;
+  outlineIsFinalized;
+  explorationId;
+  explorationSummary;
+  completed;
 
-  StoryNode.prototype.getId = function() {
-    return this._id;
-  };
+  constructor(id, title, destinationNodeIds, prerequisiteSkillIds,
+      acquiredSkillIds, outline, outlineIsFinalized, explorationId,
+      explorationSummary, completed) {
+    this.id = id;
+    this.title = title;
+    this.destinationNodeIds = destinationNodeIds;
+    this.prerequisiteSkillIds = prerequisiteSkillIds;
+    this.acquiredSkillIds = acquiredSkillIds;
+    this.outline = outline;
+    this.outlineIsFinalized = outlineIsFinalized;
+    this.explorationId = explorationId;
+    this.explorationSummary = explorationSummary;
+    this.completed = completed;
+  }
 
-  StoryNode.prototype.getTitle = function() {
-    return this._title;
-  };
+  getId() {
+    return this.id;
+  }
 
-  StoryNode.prototype.getExplorationId = function() {
-    return this._explorationId;
-  };
+  getTitle() {
+    return this.title;
+  }
 
-  StoryNode.prototype.isCompleted = function() {
-    return this._completed;
-  };
+  getExplorationId() {
+    return this.explorationId;
+  }
 
-  StoryNode.prototype.getExplorationSummaryObject = function() {
-    return this._explorationSummaryObject;
-  };
+  isCompleted() {
+    return this.completed;
+  }
 
-  StoryNode.prototype.getOutline = function() {
-    return this._outline;
-  };
+  getExplorationSummaryObject() {
+    return this.explorationSummary;
+  }
 
-  StoryNode.prototype.getOutlineStatus = function() {
-    return this._outlineIsFinalized;
-  };
+  getOutline() {
+    return this.outline;
+  }
 
-  // Static class methods. Note that "this" is not available in static
-  // contexts. This function takes a JSON object which represents a backend
-  // story python dict.
-  // TODO(ankita240796): Remove the bracket notation once Angular2 gets in.
-  /* eslint-disable dot-notation */
-  StoryNode['createFromBackendDict'] = function(storyNodeBackendObject) {
-  /* eslint-enable dot-notation */
-    return new StoryNode(
-      storyNodeBackendObject.id, storyNodeBackendObject.title,
-      storyNodeBackendObject.destination_node_ids,
-      storyNodeBackendObject.prerequisite_skill_ids,
-      storyNodeBackendObject.acquired_skill_ids,
-      storyNodeBackendObject.outline,
-      storyNodeBackendObject.outline_is_finalized,
-      storyNodeBackendObject.exploration_id,
-      storyNodeBackendObject.exp_summary_dict,
-      storyNodeBackendObject.completed
-    );
-  };
+  getOutlineStatus() {
+    return this.outlineIsFinalized;
+  }
+}
 
-  return StoryNode;
-}]);
+@Injectable({
+  providedIn: 'root'
+})
+export class ReadOnlyStoryNodeObjectFactory {
+  constructor() {}
```

Comment:
If the constructor is not needed, we can remove that right?

Reply:
Done!